help/Readme.md

```diff
@@ -0,0 +1,140 @@
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to temporary VM packer will use WinRM, or SSH connections on public IP interfaces.
+If you use build agent located in some Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case if firewall restrictions forbid connections from public addresses, you may deploy private virtual network resources, and pass them as arguments to packer, so temporary virtual machines can use private connections inside Vlan. 
+
+#### Service principal
+Packer will authorize in Azure infrastructure via Service Principal. If you want to prepare image-generation CI, or use packer manually, you need to create your SP with full read-write permissions for selected Azure subscription.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+You can use local machine as build agent, or deploy virtual machine in [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli) 
+
+Download `packer` from https://www.packer.io/downloads
+```
+$packerFolder = "C:\packer"
+Invoke-WebRequest -Uri https://releases.hashicorp.com/packer/1.6.0/packer_1.6.0_windows_amd64.zip -OutFile "$packerFolder/packer.zip" && Expand-Archive $packerFolder/packer.zip .
+```
+Or install it via Chocolately
+```
+Choco-Install -PackageName packer
+```
+
+Install Azure CLI - https://docs.microsoft.com/ru-ru/cli/azure/install-azure-cli-windows?view=azure-cli-latest&tabs=azure-cli
+```
+Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm .\AzureCLI.msi
+```
+
+Download Virtual-Envionments repository
+```
+git clone https://github.com/actions/virtual-environments.git
+```
+
+Run `packer` with specified arguments
+```
+packer build    -var "capture_name_prefix=$ResourcesNamePrefix" `
+                -var "client_id=$ClientId" `
+                -var "client_secret=$ClientSecret" `
+                -var "install_password=$InstallPassword" `
+                -var "github_feed_token=$GitHubFeedToken" `
+                -var "location=$Location" `
+                -var "resource_group=$ResourceGroup" `
+                -var "storage_account=$StorageAccount" `
+                -var "subscription_id=$SubscriptionId" `
+                -var "temp_resource_group_name=$TempResourceGroupName" `
+                -var "tenant_id=$TenantId" `
+                -var "virtual_network_name=$VirtualNetworkName" `
+                -var "virtual_network_resource_group_name=$VirtualNetworkRG" `
+                -var "virtual_network_subnet_name=$VirtualNetworkSubnet" `
+                $TemplatePath `
+```
```

Comment:
Instead of describing these parameters, I would suggest customer to use our script: https://github.com/actions/virtual-environments/blob/main/helpers/GenerateResourcesAndImage.ps1
And describe parameters of this script (they are pretty same)

We can describe remain parameters separatelly but I think this script should the main way to run packer build for customers because it automates a lot of work like storage account creation, service principal creation and etc.

Reply:
Fixed. However, I suggest to leave several parameters in `Additional Parameters` topic, because they can be used to significantly reduce image generation time.